### PR TITLE
expression: Fix wrong result of microsecond function in vectorized

### LINKED ISSUE #29244: Different `microSecond` result between non-vectorized and vectorized expression
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
drop table t;
create table t(a time(4));
insert into t values("-700:10:10.123456111");
set tidb_enable_vectorized_expression = on;
select microsecond(a) from t;
set tidb_enable_vectorized_expression = off;
select microsecond(a) from t;
```

### 2. What did you expect to see? (Required)
Same result.
```
[tidb]> select microsecond(a) from t;
+----------------+
| microsecond(a) |
+----------------+
|         123500 |
+----------------+
1 row in set (0.001 sec)
```
### 3. What did you see instead (Required)
```
[tidb]> set tidb_enable_vectorized_expression = on;
Query OK, 0 rows affected (0.000 sec)

[tidb]> select microsecond(a) from t;
+----------------+
| microsecond(a) |
+----------------+
|        -123500 |
+----------------+
1 row in set (0.001 sec)

[tidb]> set tidb_enable_vectorized_expression = off;
Query OK, 0 rows affected (0.000 sec)

[tidb]> select microsecond(a) from t;
+----------------+
| microsecond(a) |
+----------------+
|         123500 |
+----------------+
1 row in set (0.001 sec)

```

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #29244 

Problem Summary:

We should use types.Duration.MicroSecond() to get micro second.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Fix wrong result of microsecond function in vectorized
```

### CODE REVIEW

**Comment by Defined2014** [comment]
/cc @kennytm @wshwsh12 @XuHuaiyu 
PTAL

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -965,12 +965,11 @@ func (b *builtinMicroSecondSig) vecEvalInt(input *chunk.Chunk, result *chunk.Col
 	result.ResizeInt64(n, false)
 	result.MergeNulls(buf)
 	i64s := result.Int64s()
-	ds := buf.GoDurations()
 	for i := 0; i < n; i++ {
 		if result.IsNull(i) {
 			continue
 		}
-		i64s[i] = int64((ds[i] % time.Second) / time.Microsecond)
+		i64s[i] = int64(buf.GetDuration(i, 0).MicroSecond())
```
Comment (wshwsh12) [inline expression/builtin_time_vec.go:972]:
```suggestion
		i64s[i] = int64(buf.GetDuration(i, int(types.UnspecifiedFsp)).MicroSecond())
```

Comment (Defined2014) [reply expression/builtin_time_vec.go]: Done

**Review by wshwsh12 — APPROVED** [review]

**Comment by Defined2014** [comment]
/cc @xhebox 
PTAL

**Review by xhebox — APPROVED** [review]

**Pull request merged**